### PR TITLE
Upgrade rails to v4.2

### PR DESCRIPTION
**Pull request by rimenes**

### CODE REVIEW

**Comment by edmilsonefs** [comment]
Thanks, I'm using that too.

File: Gemfile [hunk 1]
```diff
@@ -16,6 +16,7 @@ gem 'will_paginate', '~> 3.0'
 group :development do
   gem 'awesome_print'
   gem 'spring'
+  gem 'web-console', '~> 2.0'
```
Comment (hugopl) [inline Gemfile]: This gem seems nice, but this addition should go in another commit, since this is about update to rails 4.2

File: bin/setup [hunk 2]
```diff
@@ -0,0 +1,29 @@
+#!/usr/bin/env ruby
```
Comment (hugopl) [inline bin/setup]: There's already a script called `first_install ` in scripts directory that do something like that, also this has nothing to do with rails 4.2 update ;-)

File: config/boot.rb [hunk 3]
```diff
@@ -1,4 +1,3 @@
-# Set up gems listed in the Gemfile.
 ENV['BUNDLE_GEMFILE'] ||= File.expand_path('../../Gemfile', __FILE__)
 
-require 'bundler/setup' if File.exist?(ENV['BUNDLE_GEMFILE'])
+require 'bundler/setup' # Set up gems listed in the Gemfile.
```
Comment (hugopl) [inline config/boot.rb]: Is this changed in rails 4.2? Do yo know why?

File: bin/setup (hunk 2 above)
Comment (rimenes) [inline bin/setup]: the bin/setup file is the "rails  way" to setup projects.... i'd create a ticket to merge scritpts/first_install into bin/setup after this PR got merged

File: config/environments/production.rb [hunk 4]
```diff
@@ -16,11 +16,13 @@
...
-  # For large-scale production use, consider using a caching reverse proxy like nginx, varnish or squid.
+  # For large-scale production use, consider using a caching reverse proxy like
+  # NGINX, varnish or squid.
   # config.action_dispatch.rack_cache = true
 
-  # Disable Rails's static asset server (Apache or nginx will already do this).
-  config.serve_static_assets = false
+  # Disable serving static files from the `/public` folder by default since
+  # Apache or NGINX already handles this.
+  config.serve_static_files = ENV['RAILS_SERVE_STATIC_FILES'].present?
```
Comment (hugopl) [inline config/environments/production.rb]: We would move this config to reviewit.yml, but later... better keep it as is meanwhile then move it to reviewit.yml in another PR.

File: Gemfile (hunk 1 above)
Comment (rimenes) [inline Gemfile]: You're right. This gem is suggested in rails 4.2, but not a dependecy.

File: config/environments/production.rb [hunk 5]
```diff
@@ -29,20 +31,22 @@
...
+  # config.action_dispatch.x_sendfile_header = 'X-Accel-Redirect' # for NGINX
 
   # Force all access to the app over SSL, use Strict-Transport-Security, and use secure cookies.
   # config.force_ssl = true
 
-  # Set to :debug to see everything in the log.
-  config.log_level = :info
+  # Use the lowest log level to ensure availability of diagnostic information
+  # when problems arise.
+  config.log_level = :debug
```
Comment (hugopl) [inline config/environments/production.rb]: Debug log_level on production!?

File: config/environments/production.rb [hunk 6]
```diff
@@ -67,9 +71,6 @@
   # Send deprecation notices to registered listeners.
   config.active_support.deprecation = :notify
 
-  # Disable automatic flushing of the log to improve performance.
-  # config.autoflush_log = false
```
Comment (hugopl) [inline config/environments/production.rb]: Was this config value removed from rails?

File: config/environments/test.rb [hunk 7]
```diff
@@ -12,8 +12,8 @@
   # preloads Rails for running tests, you may have to set it to true.
   config.eager_load = false
 
-  # Configure static asset server for tests with Cache-Control for performance.
-  config.serve_static_assets  = true
+  # Configure static file server for tests with Cache-Control for performance.
+  config.serve_static_files   = true
```
Comment (hugopl) [inline config/environments/test.rb]: Sorry... but are you sure this really changed from assets to files? I didn't see the rails 4.2 changelog yet.

File: bin/setup (hunk 2 above)
Comment (hugopl) [inline bin/setup]: So we can merge scripts/first_install into setup, but in another PR, better keep this PR only with rails 4.2 update :-)

File: config/environments/test.rb (hunk 7 above)
Comment (rimenes) [inline config/environments/test.rb]: Yep. Commit deprecating serve_static_assets https://github.com/rails/rails/commit/0082f02d1efcfd65b4c7f0f80db4356f82271d70

File: config/boot.rb (hunk 3 above)
Comment (rimenes) [inline config/boot.rb]: Yes. Don't know the real reason, but AFAIK "require 'bundler/setup' "should always run in the initialization process. If the Gemfile is not found the app won't boot up any way.

File: config/environments/production.rb (hunk 5 above)
Comment (rimenes) [inline config/environments/production.rb]: That's the new default. Certainly it won't make sense in lots of deployments. Gonna put the info level back.

File: config/environments/production.rb (hunk 6 above)
Comment (rimenes) [inline config/environments/production.rb]: It's still in rails and defaults to true. It's just not part of the 4.2 production.rb template. btw it's was commented out in the current version.

**Comment by hugopl** [comment]
I squashed all commits in a single one and pushed it into 4c84c5133c8789021223e413b8eb41d50b40328f, so closing this PR.
